### PR TITLE
Add some fields and update sdk API

### PR DESCRIPTION
**Pull request by i1i1**
This pr adds lots of new fields for node.

### CODE REVIEW

File: src/farmer.rs [hunk 1]
```diff
@@ -90,21 +92,24 @@ impl PlotDescription {
...
-struct DSNRecordsCacheSize(NonZeroUsize);
-
-impl Default for DSNRecordsCacheSize {
-    fn default() -> Self {
-        Self(NonZeroUsize::new(32678).unwrap())
+mod builder {
+    use libp2p_core::Multiaddr;
+
+    /// Technical type which stores all
+    #[derive(Debug, derive_builder::Builder)]
```
Comment (ozgunozerk) [inline src/farmer.rs:99]: that's interesting 👀  (derive_builder)

**Review by ozgunozerk — APPROVED** [review]
I really liked the flexibility and the organization of the new approach. Wonderful work @i1i1 🥳 

I have a small concern, which is not a blocker for this PR, so I'm approving.

Here is the concern: (examples/complete.rs -> line 13)
```rust
.network(NetworkBuilder::new().name("i1i1").build())
```
this API seemed a bit weird to me. It's builder inside a builder (builderception).
I currently do no have a better alternative in my mind, but just feeling like we can make this a bit easier and simpler for the end-developer. 
Dropping this comment, just to spark some brainstorming :)

LGTM otherwise!

File: src/node/mod.rs [hunk 2]
```diff
@@ -35,6 +35,167 @@ pub use sc_state_db::Constraints;
...
+        /// Maximum allowed subscriptions per rpc connection
+        #[builder(default = "1024")]
+        #[derivative(Default(value = "1024"))]
+        pub max_subs_per_conn: usize,
+        /// Maximum size of the output buffer capacity for websocket connections.
+        #[builder(setter(strip_option), default)]
+        pub ws_max_out_buffer_capacity: Option<usize>,
+    }
+
+    impl RpcBuilder {
```
[the remark this answers is not in the document]
Comment (i1i1) [reply src/node/mod.rs:125]: `RpcBuilder` is created by `derive_builder`, I just add some methods

**Comment by i1i1** [comment]
> this API seemed a bit weird to me. It's builder inside a builder (builderception).
> I currently do no have a better alternative in my mind, but just feeling like we can make this a bit easier and simpler for the end-developer.
> Dropping this comment, just to spark some brainstorming :)

Yeah, so that it would be subdivided. It looks better for me. WDUT?

**Pull request merged**